src/model_selection/scorers.ts

```diff
@@ -0,0 +1,62 @@
+/**
+*  @license
+* Copyright 2022, JsData. All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+* ==========================================================================
+*/
+
+import { Scikit1D, Scikit2D } from '../types'
+import { convertToTensor1D } from '../utils'
+import { tf } from '../shared/globals'
+type Scalar = tf.Scalar
+type Tensor1D = tf.Tensor1D
+
+export function negMeanAbsoluteError(
+  this: {
+    predict(X: Scikit2D): Tensor1D
+  },
+  X: Scikit2D,
+  y: Scikit1D
+): Scalar {
+  return tf.tidy(() => {
+    y = convertToTensor1D(y)
+    const yPred = this.predict(X)
+    return tf.metrics.meanAbsoluteError(y, yPred).neg() as Scalar
+  })
+}
+
+export function negMeanSquaredError(
+  this: {
+    predict(X: Scikit2D): Tensor1D
+  },
+  X: Scikit2D,
+  y: Scikit1D
+): Scalar {
+  return tf.tidy(() => {
+    y = convertToTensor1D(y)
+    const yPred = this.predict(X)
+    return tf.metrics.meanSquaredError(y, yPred).neg() as Scalar
+  })
+}
+
+export function accuracy(
+  this: {
+    predict(X: Scikit2D): Tensor1D
+  },
+  X: Scikit2D,
+  y: Scikit1D
+): Scalar {
+  return tf.tidy(() => {
+    y = convertToTensor1D(y)
+    const yPred = this.predict(X)
+    return tf.equal(y, yPred).sum().div(y.shape[0]) as Scalar
```

Comment:
Is there a reason why you aren't just using the functions in metrics? They are a thin wrapper around the tfjs api, and they already do the conversions to / from Scikit1D as well as some sanity checks on the input. Like this whole function could just be

```
export function accuracy(
  this: {
    predict(X: Scikit2D): Tensor1D
  },
  X: Scikit2D,
  y: Scikit1D
): Scalar {
  return tf.tidy(() => {
    return tf.scalar(metrics.accuracyScore(y, this.predict(X))) as Scalar
  })
}
```

Right now, all the functions in metrics return a `number` but if you want the result to be a `Scalar` instead of a `number`, we can simply change those functions to optionally return a Scalar. Like have a function call that looks like

`metrics.accuracyScore(labels, predictions) // returns number`
`metrics.accuracyScore(labels, predictions, true) // returns Scaler`

Reply:
If this doesn't break anything sure. In my opinion `Scalar` is the datatype of choice because it allows to use sweet sweet backpropagation to optimize hyperparameters.

Once the changes are made to `metrics.ts`, we can even use a `makeScorer` as they do in `sklearn`.

But maybe that's a story for another commit? The commit is growing bigger and bigger touching more and more files.